types/specification.d.ts

```diff
@@ -0,0 +1,105 @@
+/** Stamp Specification v1.6 */
+declare namespace Specification {
+  /** Base type for `properties` */
+  type PropertyMap = { [s: string]: any; };
+
+  /** Base type for `methods` */
+  type MethodMap = { [s: string]: (...args: Array<any>) => any; };
+
+  /** Extract the instance type from Factory, Descriptor (or Instance) */
+  type StampType<I> = I extends Factory<infer I> ? I
+                    : I extends Descriptor<infer I> ? I
+                    : I extends Instance ? I
+                    : never;
+
+  /** A composable object (`stamp` factory or Descriptor) */
+  type Composable<I, U = StampType<I>> = Factory<U> | Descriptor<U>;
+
+  /** Base `stamp` instance */
+  type Instance = { [s: string]: any; };
+
+  /**
+   * A factory function that will produce new objects using the
+   * prototypes that are passed in or composed.
```

Comment:
"using the prototypes that are passed" ? Can I suggest rewording of the comment: "using the matadata that is attached to `.compose`"

Reply:
Blame the one who did the original `@types/stampit` package :)

I first tried to stick with the terms used in Stamp Specification and later used description from the `@types/stampit` package (which are often confusing IMHO)

Can you edit the files in this PR?

Reply:
Blame them! :)

I can't edit files in this PR since it's your repo. :)

However, how about I give you write access to the stampit repo?